### PR TITLE
The alternating group is generated by 3-cycles

### PR DESCRIPTION
**Pull request by tirix**
Follow-up for #3494 and #3518.

The main result is that the alternating group is generated by 3-cycles. This is proved using induction on words of even length (special theorem for that, I'm unsure whether it will have other uses).
Four theorems moved from GS's mathbox to the main part of set.mm.

### CODE REVIEW

File: set.mm [hunk 1]
```diff
@@ -460080,6 +460128,40 @@ Class abstractions (a.k.a. class builders)
...
+      sylancl eqeltrd exlimddv ) BAEZCBFZBGHIZUAZBCJZUBZDVCZJZKZWDLZAFDWBWDMHIW
+      GDUCWBWDWDNOZMHWBWIWDWBWDSWBBSFZWDSFWAVSWJVTWAWJGSFZGUDFZWKPGUEUFBGUGUHQZ
+      BWCUIUJUKULWBWITZMTZKWIMKWBBNOZWNWOWBWJVTWPWNKWMVSVTWAUMBCUNUOWAVSWPWOKVT
+      WAWPGWOWAWLWPGKPBGUPUQURRQUSWIMUTVAVBDWDVDVAWBWGVEZWHWEAWQWHWFLWEWQWDWFWB
+      WGVFZVGWEDVHVIRWQWFAEWEWFFWEAFWQWFBAWQWFWDBWRWQBWCVJVKVSVTWAWGVLVMDVNWFAW
+      EVOVPVQVR $.
+
+    $( The other element of a pair is not the known element.  (Contributed by
+       Thierry Arnoux, 26-Aug-2017.) $)
+    unidifsneq $p |- ( ( X e. P /\ P ~~ 2o ) -> U. ( P \ { X } ) =/= X ) $=
```
Comment (icecream17) [inline set.mm:460150]: maybe
```suggestion
    unidifsnne $p |- ( ( X e. P /\ P ~~ 2o ) -> U. ( P \ { X } ) =/= X ) $=
```

**Review by icecream17 — APPROVED** [review]

File: set.mm (hunk 1 above)
Comment (tirix) [reply set.mm:460150]: Yes indeed! Change done.

**Comment by savask** [comment]
A cool result! Just a few comments:

1. One in principle can remove `3 <_ N` from the assumptions of `cyc3genpm` since Alt(1) and Alt(2) are trivial and hence generated by an empty set.
2. I think alternating groups deserve their own notation.
3. I also think "a subgroup generated by a set" deserves its own notation as well.

**Comment by tirix** [comment]
> 1. One in principle can remove `3 <_ N` from the assumptions of `cyc3genpm` since Alt(1) and Alt(2) are trivial and hence generated by an empty set.

Yes! I might try to remove that condition.

> 2. I think alternating groups deserve their own notation.

There is already `pmEven` which I'm using. `` ( pmEven ` D ) `` is the set of even permutation on the set `D`, i.e. _Alt(n)_ if `D` has _n_ elements. Note that this is just a set, it does not have the group structure, but it is easy to inherit from. 
The naming is not great, though. I would also prefer something that would sound more like "alternating group".

> 3. I also think "a subgroup generated by a set" deserves its own notation as well.

Yes, possibly. For the notation, I've got inspiration from [`psgnfitr`](https://us.metamath.org/mpeuni/psgnfitr.html): no special notation.
I think those are the only two examples for now?

Next up is conjugates, I have [`sylow2b`](https://us.metamath.org/mpeuni/sylow2b.html) as a model for the statement, which also does not have a special notation.

**Comment by savask** [comment]
>  I think those are the only two examples for now?

I guess there are also theorems about cyclic groups like [ablsimpg1gend](https://us.metamath.org/mpeuni/ablsimpg1gend.html), but these don't need words to work with generators.

File: set.mm [hunk 2]
```diff
@@ -460080,6 +460128,40 @@ Class abstractions (a.k.a. class builders)
       EBCRUIUOUEUBUFMUG $.
   $}
 
+  ${
+    $d A x $.  $d P x $.  $d X x $.
+    $( The other element of a pair is an element of the pair.  (Contributed by
+       Thierry Arnoux, 26-Aug-2017.) $)
+    unidifsnel $p |- ( ( P C_ A /\ X e. P /\ P ~~ 2o )
+                  -> U. ( P \ { X } ) e. A ) $=
```
Comment (avekens) [inline set.mm:460136]: Interesting: I was not aware of the idiom ` U. ( P \ { X } ) ` for "the other element" of a pair ` P ` until now. Unfortunately, this is nowhere explicitly explained in set.mm, only implicitly by ~en2eleq.

**Review by avekens — APPROVED** [review]

File: set.mm (hunk 2 above)
Comment (tirix) [reply set.mm:460136]: This is thanks to Stefan O'Rear I believe. Very clever!
I've made a new version which I think is slightly simpler, so possibly more reusable.

**Comment by tirix** [comment]
> I might try to remove the N≥3 condition.

Done in the last commit!

**Pull request merged**